lightning/src/ln/channelmanager.rs

```diff
@@ -7386,6 +7407,7 @@ impl Readable for ChannelDetails {
 			(35, inbound_htlc_maximum_msat, option),
 			(37, user_channel_id_high_opt, option),
 			(39, feerate_sat_per_1000_weight, option),
+			(41, channel_shutdown_state, required),
```

Comment:
`required` means we'll fail to deserialize any `ChannelDetails` that were serialized prior to this PR, which isn't ideal. This could either be made into an `Option`al field, or we could have it default to `NotShuttingDown` and document when the field may be inaccurate.

Reply:
I see, in that case probably optional is the right setting here.

Reply:
Yeah, missed this comment on the first pass! Definitely agree with using `option` here.